src/Nethermind/Nethermind.Init/Steps/LoadGenesisBlock.cs

```diff
@@ -97,7 +97,7 @@ private void ValidateGenesisHash(Keccak? expectedGenesisHash)
             if (_api.WorldState is null) throw new StepDependencyException(nameof(_api.WorldState));
             if (_api.BlockTree is null) throw new StepDependencyException(nameof(_api.BlockTree));
 
-            BlockHeader genesis = _api.BlockTree.Genesis!;
+            BlockHeader genesis = _api.BlockTree.Genesis ?? throw new NullReferenceException("Genesis block is null");
```

Comment:
You should never throw NullReferenceException: https://stackoverflow.com/a/22453825

Reply:
Do you suggest to use `ArgumentNullException`?